### ISSUE TITLE
Being able to turn off not important logs

### ISSUE THREAD

**Issue by xonvanetta** [kuskoman/logstash-exporter#151]
Currently all logs are logged with the standard log package, would be nice to be able to disable all the `info` logs if we call it that.

I could make a PR but since the flavor or logging these days seem to be split, I would still use logrus and could do a PR if that's OK with you.

**Comment by kuskoman** [comment]
@xonvanetta some time ago I created a PR/branch (I will look for it in a moment) with some of structured logs, but I remember I encountered problems that I haven't dealt before and just abandoned it.
Also, I would prefer using [slog](https://pkg.go.dev/golang.org/x/exp/slog) package, which will probably become a part of the standard library, than adding external dependency

**Comment by kuskoman** [comment]
I'm open to propositions on how logging should be done there, however before implementing it please present examples, so we can elaborate on its consequences

**Comment by xonvanetta** [comment]
Never worked with slog but I can write some examples, probably even in an fork to verify behavior and also to test

**Comment by xonvanetta** [comment]
So pushed a POC branch on my fork, also don't know if i got every log correct as in which level it should be logged on but that's up for discussion, same for default log level and default text formatter or json formatter. 

Lil examples here from the POC:
https://github.com/xonvanetta/logstash-exporter/commit/6dc5d2f1dfc643d85ab7a76611de7810f69072fa#diff-1fabb01a2607990d88b8259081c988c3bb7b601909c2484573ead734f3e2c56fL76-R80
```
time=2023-07-18T12:35:51.553+02:00 level=ERROR msg="executor failed" name=nodeinfo duration=992.183µs err="Get \"http://localhost:9600\": dial tcp [::1]:9600: connect: connection refused"
```

https://github.com/xonvanetta/logstash-exporter/commit/6dc5d2f1dfc643d85ab7a76611de7810f69072fa#diff-87a14da5a23cb0add260fbc88283bec5f8b2caeea3aefb8794f9a4c0277bb061L177-R178
```
time=2023-07-18T12:39:14.201+02:00 level=DEBUG msg="collecting pipeline plugin stats for pipeline" pipelineID=beats "plugin type"=output name=opensearch id=36b2dd9dbac4bb4e07e8755b4eba1703f59d76fee0c8188b221a5de79eced88f
```

What more kind of examples would you like to have? Like how to use it and how it's being used?

**Comment by kuskoman** [comment]
@xonvanetta I like it so far, feel free to keep going with it. also, please tell me if you need any help with converting the project to use slog, so I may help

**Comment by xonvanetta** [comment]
I fixed some small typos but we could move the more fine detail discussion to the PR, otherwise I haven't had any problems so far with it

**Comment by kuskoman** [comment]
@xonvanetta I just released v1.5.0 version that contains the slog
